packages/lib/slots.ts

```diff
@@ -132,10 +191,40 @@ function buildSlots({
      * @calcom/web:dev: 2022-11-06T03:00:00-04:00
      * ...
      */
-    slots.push({
-      userIds: item.userIds,
-      time: getTime(item.startTime),
-    });
+    const timeNum = parseInt(time);
+    const slotTime = getTime(timeNum);
+    // Skip slots that are before the day's start or after its end
+    if (timeNum < dayStart) continue;
+
+    let slotData: {
+      time: Dayjs;
+      userIds?: number[];
+      away?: boolean;
+      fromUser?: IFromUser;
+      toUser?: IToUser;
+      reason?: string;
+      emoji?: string;
+      busy?: boolean;
+    } = {
+      userIds: isAvailable ? isAvailable.userIds : undefined,
+      time: slotTime,
+      busy: !isAvailable, // Mark non-available slots as busy
+    };
+
+    // Always add OOO data if it exists, regardless of availabilityy
+    if (dateOutOfOfficeExists) {
+      const { toUser, fromUser, reason, emoji } = dateOutOfOfficeExists;
+      slotData = {
+        ...slotData,
+        away: true,
+        ...(fromUser && { fromUser }),
+        ...(toUser && { toUser }),
+        ...(reason && { reason }),
+        ...(emoji && { emoji }),
+      };
+    }
+
+    slots.push(slotData);
```

Comment:
- this can be refactored into much fewer lines of code
- `const { toUser, fromUser, reason, emoji } = dateOutOfOfficeExists;` I see this in multiple places. you can either destructure `dateOutOfOfficeExists` at the top once or do `dateOutOfOfficeExists?.fromUser`, etc.